src/rospkg/rospack.py

```diff
@@ -230,19 +238,39 @@ def get_depends(self, name, implicit=True):
             if name in self._depends_cache:
                 return self._depends_cache[name]
 
-            # take the union of all dependencies
-            names = [p.name for p in self.get_manifest(name).depends]
-
             # assign key before recursive call to prevent infinite case
             self._depends_cache[name] = s = set()
 
+            # take the union of all dependencies
+            names = None
+            try:
+                names = [p.name for p in self.get_manifest(name).depends]
+            except ResourceNotFound as e:
+                self._depends_unavailable.append(name)
+                print("[WARN] Depended package not available on your environment: {0}".format(str(e)))
```

Comment:
It don't think this should print a warning. It should be the code handling the exception to print an appropriate message containing all the unavailable package names.